### PR TITLE
Clean up supported platform descriptions

### PR DESCRIPTION
**Pull request by stuartmorgan**
Fixes inconsistencies in the listings:
- The initial listing now includes both Google-tested and best-effort support tiers for all platforms; it included both for some platforms, but only Google-tested for others.
- Fixes the Unsupported platforms section for Android; it was listing best-effort platforms as unsupported.
- Adds a missing version number to the Linux best-effort entry.
- Adds 32-bit Linux to the Unsupported section for consistency with the Windows entry.
- Uses ranges for Android versions instead of listing each one, as we do for iOS.

Other minor changes:
- Explicitly lists 64-bit as a requirement for Windows in the main listing, instead of relying on people finding the note about 32-bit in the Unsupported section.
- Explicitly lists 64-bit as a requirement for Linux in the main listing.
- Adds OS version numbers for macOS, since it's hard to remember the name->version mapping for every macOS version.
- Adds an Android OS version number in the mail listing, to make it easy to map the support to user-visible OS numbers instead of needing to know the SDK->OS version mapping.

Fixes https://github.com/flutter/flutter/issues/106301

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by stuartmorgan** [comment]
There's also an inconsistency in the table ordering: the main table uses alphabetical ordering, while the others use: {mobile}, {desktop in most-to-least market share order}. Should we standardize on one or the other?

**Review by sfshaza2 — COMMENTED** [review]
@stuartmorgan, this mostly looks great! But why did you remove the note about the dev channel? The 2.8 release wasn't that long ago.

File: src/development/tools/sdk/release-notes/supported-platforms.md [hunk 1]
```diff
@@ -75,28 +59,29 @@ Note that Android SDK 20 is covered by
 testing Android SDK 19, as the differences
 between the two platform versions are minimal.
```
Comment (Hixie) [inline src/development/tools/sdk/release-notes/supported-platforms.md:60]: we could just remove this paragraph, i'm not sure what it really adds

**Comment by Hixie** [comment]
> why did you remove the note about the dev channel? The 2.8 release wasn't that long ago.

This is a weird page on which to draw attention to it, I support not mentioning it here. We have mentions of it in lots of other places still.

**Review by Hixie — APPROVED** [review]
Standardising on one order seems reasonable.

The changes all LGTM.

fyi @timsneath in case this suggests any policy changes we should be considering (the changes here are for accuracy and represent what we are actually doing today, but that may be in conflict with what we are intending to do)

LGTM

**Review by johnpryan — APPROVED** [review]

**Comment by Hixie** [comment]
I'd go with alphabetical.

**Comment by sfshaza2** [comment]
lgtm

**Pull request merged**